server2/Arcadia.Assistant/Arcadia.Assistant.AnniversaryFeed/AnniversaryFeed.cs

```diff
@@ -0,0 +1,120 @@
+namespace Arcadia.Assistant.AnniversaryFeed
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Fabric;
+    using System.Linq;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    using Autofac.Features.OwnedInstances;
+
+    using Contracts;
+
+    using CSP;
+    using CSP.Model;
+
+    using Microsoft.EntityFrameworkCore;
+    using Microsoft.Extensions.Logging;
+    using Microsoft.ServiceFabric.Services.Communication.Runtime;
+    using Microsoft.ServiceFabric.Services.Remoting.Runtime;
+    using Microsoft.ServiceFabric.Services.Runtime;
+
+    using UserFeeds.Contracts;
+
+    /// <summary>
+    ///     An instance of this class is created for each service instance by the Service Fabric runtime.
+    /// </summary>
+    public class AnniversaryFeed : StatelessService, IAnniversaryFeed
+    {
+        private readonly Func<Owned<CspEmployeeQuery>> employeeQuery;
+        private readonly ILogger logger;
+
+        public AnniversaryFeed(
+            StatelessServiceContext context, Func<Owned<CspEmployeeQuery>> employeeQuery,
+            ILogger<AnniversaryFeed> logger)
+            : base(context)
+        {
+            this.employeeQuery = employeeQuery;
+            this.logger = logger;
+        }
+
+        public string ServiceType => Constants.ServiceType;
+
+        public async Task<FeedItem[]> GetItems(
+            DateTime startDate, DateTime endDate, CancellationToken cancellationToken)
+        {
+            using var query = this.employeeQuery();
+            return (await query.Value.Get()
```

Comment:
why don't you request Employees service instead? You duplicate some logic (Full names, etc)

Reply:
We have no required data and interface in IEmployee service. Could we extend this service?

Reply:
where is a query interface, isn't? You can extend it.

Reply:
Employee service used instead direct csp calls